### ISSUE TITLE
Support XDG Base directory specification

### ISSUE THREAD

**Issue by afranke** [jgm/pandoc#3582]
The manual currently says that if --data-dir is not specified the default value is `~/.pandoc/`.

The [XDG Base directory spec](https://standards.freedesktop.org/basedir-spec/basedir-spec-latest.html) defines `$XDG_DATA_HOME` (`~/.local/share` by default) and Pandoc should take this into account.

For compatibility all three locations could be searched: first `$XDG_DATA_HOME/pandoc/`, then `~/.local/share/pandoc/` and finally `~/.pandoc/`.

**Comment by jgm** [comment]
We currently use the standard Haskell function `getAppUserDataDirectory` to get this.

http://hackage.haskell.org/package/directory-1.3.3.2/docs/System-Directory.html#v:getAppUserDataDirectory

Looking at the docs, I see that in directory >= 1.2.3.0, they also have `getXdgDirectory` and `getXdgDirectoryList`.

I'd have to think carefully about what would be necessary to make the change. It might be quite easy, but we should do it in a backwards-compatible way (searching `~/.pandoc` too).

**Comment by barskern** [comment]
Perhaps the easiest way of resolving this would be to firstly check the XDG directory, and if it does not exist, it will fall back to the old solution? As I understand it, `pandoc` will not write to that directory, but the user can specify custom templates there, hence as long as `pandoc` doesn't need to chose where to write, it shouldn't be backwards incompatible.

**Comment by jgm** [comment]
Ole Martin Ruud <notifications@github.com> writes:

> Perhaps the easiest way of resolving this would be
> to firstly check the XDG directory, and if it does
> not exist, it will fall back to the old solution? As

This is what I've done.

**Comment by 0xdevalias** [comment]
> Perhaps the easiest way of resolving this would be to firstly check the XDG directory, and if it does not exist, it will fall back to the old solution? As
> This is what I've done.

In case anyone was wondering where this is implemented, it's here:

https://github.com/jgm/pandoc/blob/fd496848d8a9799edc5095ed3134522a9c14ad11/src/Text/Pandoc/Data.hs#L228-L245

(Context: I was trying to find a canonical list of files/paths pandoc uses within `XDG_DATA_HOME`)

**Comment by jgm** [comment]
See the `data/` directory in the source.
All the subdirectories there should be overridable in XDG_DATA_HOME.

**Comment by 0xdevalias** [comment]
> See the `data/` directory in the source.

@jgm I did see that, but it doesn't include everything. For example, there is no `custom` or `filters` folders there. From my skimming through the docs I suspect there are others too.

eg. From https://github.com/0xdevalias/dotfiles/commit/1eebd2292942478a5e042b513a4824b0d913a391

```
[local] pandoc: create placeholder DATADIR folder structure (for at least some of the folders here)
- https://github.com/jgm/pandoc/issues/3582#issuecomment-2586099054
- https://pandoc.org/MANUAL.html#option--data-dir
  - DATADIR
    - $XDG_DATA_HOME/pandoc
    - $HOME/.local/share/pandoc
    - $HOME/.pandoc
- https://pandoc.org/MANUAL.html#option--filter
  - $DATADIR/filters
- https://pandoc.org/MANUAL.html#option--lua-filter
  - $DATADIR/filters
- https://pandoc.org/MANUAL.html#custom-readers-and-writers
  - $DATADIR/custom
- https://pandoc.org/MANUAL.html#option--defaults
  - $DATADIR/defaults
- https://pandoc.org/MANUAL.html#option--metadata-file
  - $DATADIR/metadata
- https://pandoc.org/MANUAL.html#templates
  - $DATADIR/templates/default.*FORMAT*
- https://pandoc.org/MANUAL.html#option--template
  - $DATADIR/templates
- https://pandoc.org/MANUAL.html#option--abbreviations
  - $DATADIR/abbreviations
- https://pandoc.org/MANUAL.html#styling-the-slides
  - $DATADIR/s5
  - $DATADIR/slidy
  - $DATADIR/slideous
- etc
```

**Comment by jgm** [comment]
> Would it make sense to maintain that structure in the source repo, even for empty folders

I think this would *only* be useful as a form of documentation (since there are no system data files to override in these subdirectories).  And it's not great as a form of documentation, since it's a pretty rare user who will be looking at the source code to figure this out.  It might be worth documenting this somewhere, though.

**Comment by 0xdevalias** [comment]
> I think this would _only_ be useful as a form of documentation

Agreed.

> And it's not great as a form of documentation

Agreed.

> It might be worth documenting this somewhere, though.

At the risk of potentially slightly raising maintenance burden if/when these things change, I definitely would have found it useful the other day. But also not too sure how often this need would arise for people in a 'grand scheme' type way (I wanted to setup the folder structure for my dotfiles), as opposed to just looking for the specific individual thing they need in the moment.

One thing I did notice while skimming the manual was that some entries use the convention of mentioning `$DATADIR`, and those ones were relatively easy to find even across the different manual entries. But then there were also a bunch that just mentioned something like 'user data directory' or similar, and I think some others that didn't necessarily follow that pattern and/or may not have been documented (or at least found during my skim through the manual) at all.

I also noticed that some mentions of 'user data directory' link back to `--data-dir`, but not all, often like this:

> (see [--data-dir](https://pandoc.org/MANUAL.html#option--data-dir))

Like in:

- https://pandoc.org/MANUAL.html#option--reference-doc

For Docx it mentions:

> If no reference docx is specified on the command line, pandoc will look for a file reference.docx in the user data directory (see [--data-dir](https://pandoc.org/MANUAL.html#option--data-dir)).

And similar for ODT:

> If no reference ODT is specified on the command line, pandoc will look for a file reference.odt in the user data directory (see [--data-dir](https://pandoc.org/MANUAL.html#option--data-dir)).

But then PowerPoint doesn't mention that part at all.
